extensions/scarb-snforge-test-collector/src/compilation/test_collector.rs

```diff
@@ -0,0 +1,253 @@
+        })
+        .collect();
+
+    let sierra_program = db
+        .get_sierra_program_for_functions(z)
+        .to_option()
+        .context("Compilation failed without any diagnostics")
+        .context("Failed to get sierra program")?;
+
+    let collected_tests: Vec<TestCaseRaw> = all_tests
+        .into_iter()
+        .map(|(func_id, test)| {
+            (
+                format!(
+                    "{:?}",
+                    FunctionLongId {
+                        function: ConcreteFunction {
+                            generic_function: GenericFunctionId::Free(func_id),
+                            generic_args: vec![]
+                        }
+                    }
+                    .debug(db)
+                ),
+                test,
+            )
+        })
+        .collect_vec()
+        .into_iter()
+        .map(|(test_name, config)| TestCase {
+            name: test_name,
+            available_gas: config.available_gas,
+            ignored: config.ignored,
+            expected_result: config.expectation,
+            fork_config: config.fork_config,
+            fuzzer_config: config.fuzzer_config,
+        })
+        .collect();
+
+    let sierra_program = replace_sierra_ids_in_program(db, &sierra_program);
+
+    validate_tests(sierra_program.clone(), &collected_tests, builtins)?;
+
+    Ok((sierra_program, collected_tests))
+}
+
+// inspired with cairo-lang-compiler/src/project.rs:49 (part of setup_single_project_file)
+fn insert_lib_entrypoint_content_into_db(
+    db: &mut RootDatabase,
+    crate_name: &str,
+    crate_root: &Path,
+    lib_content: &str,
+) -> CrateId {
+    let main_crate_id = db.intern_crate(CrateLongId::Real(SmolStr::from(crate_name)));
+    db.set_crate_config(
+        main_crate_id,
+        Some(CrateConfiguration::default_for_root(Directory::Real(
+            crate_root.to_path_buf(),
+        ))),
+    );
+
+    let module_id = ModuleId::CrateRoot(main_crate_id);
+    let file_id = db.module_main_file(module_id).unwrap();
+    db.as_files_group_mut()
+        .override_file_content(file_id, Some(Arc::new(lib_content.to_string())));
+
+    main_crate_id
+}
+
+fn validate_tests(
+    sierra_program: Program,
+    collected_tests: &Vec<TestCaseRaw>,
+    ignored_params: &[&str],
+) -> Result<(), anyhow::Error> {
+    let casm_generator = match SierraCasmGenerator::new(sierra_program) {
+        Ok(casm_generator) => casm_generator,
+        Err(e) => panic!("{}", e),
+    };
+    for test in collected_tests {
+        let func = casm_generator.find_function(&test.name)?;
+        let mut filtered_params: Vec<String> = Vec::new();
+        for param in &func.params {
+            let param_str = &param.ty.debug_name.as_ref().unwrap().to_string();
+            if !ignored_params.contains(&param_str.as_str()) {
+                filtered_params.push(param_str.to_string());
+            }
+        }
```

Comment:
Removing it, the filtered params were not used anywhere 🤷 
cc @MaksymilianDemitraszek @cptartur